.github/workflows/cluster_endtoend_vreplication_basic_self_hosted.yml

```diff
@@ -0,0 +1,37 @@
+# DO NOT MODIFY: THIS FILE IS GENERATED USING "make generate_ci_workflows"
+
+name: Cluster Self Hosted (vreplication_basic)
+on: [push, pull_request]
+
```

Comment:
I think that it would not help at all. Since we have only 4 runners setup, in case there are more jobs they will be queued up. Also, I salted the name of the docker container and image with the `GITHUB_SHA` for the reason that it allows us to run the same workflow with two different SHAs in parallel. Am I missing some use case that concurrency will allow us to solve?

Reply:
The concurrency feature will allow us to limit the number of jobs in the queue and thus speed up the build process. Let's assume a new commit triggers `n` new builds that run across our fleet of `4` servers. We create a push a new commit on a pull request `p`, this triggers `n` builds, if we create a push a second new commit on top of the first, this will trigger an additional `n` builds. Now our queue is `2n` long, minus the builds that might have already finished. To alleviate the size of the queue getting too long as we add a new commit to `p`, we can use the concurrency feature. The concurrency feature will cancel all existing builds related to `p` in the queue if new builds are added.
We already integrate this feature for most of our other builds: https://github.com/vitessio/vitess/pull/8546

Reply:
Yes this seems like a great idea. Thankyou! I'll do that too